File: frontend/src/types/menuPermission.ts
// E:\Audit\AMS\frontend\src\types\menuPermission.ts

export type MenuLookup = {
  id: number;
  menu_key: string;
  menu_title: string;
  route_path: string | null;
  icon: string | null;
  permission_key: string | null;
  sort_order: number;
  is_active: boolean;
};

export type MenuPermission = {
  id: number;
  menu_id: number;
  permission_id: number;
  is_active: boolean;

  menu_key?: string | null;
  menu_title?: string | null;
  route_path?: string | null;

  permission_key?: string | null;
  resource_type?: string | null;
  resource_key?: string | null;
  action?: string | null;
  description?: string | null;

  created_by: string | null;
  updated_by: string | null;
  created_at: string;
  updated_at: string;
};

export type MenuPermissionPayload = {
  menu_id: number;
  permission_id: number;
};

export type MenuPermissionMessageResponse = {
  message: string;
  data: MenuPermission | null;
};

File: backend/app/schemas/menu_permission.py
# E:\Audit\AMS\backend\app\schemas\menu_permission.py

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MenuPermissionCreate(BaseModel):
    menu_id: int
    permission_id: int


class MenuPermissionMenuResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    menu_key: str
    menu_title: str
    route_path: str | None = None
    icon: str | None = None
    permission_key: str | None = None
    sort_order: int
    is_active: bool


class MenuPermissionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    menu_id: int
    permission_id: int
    is_active: bool

    menu_key: str | None = None
    menu_title: str | None = None
    route_path: str | None = None

    permission_key: str | None = None
    resource_type: str | None = None
    resource_key: str | None = None
    action: str | None = None
    description: str | None = None

    created_by: str | None = None
    updated_by: str | None = None
    created_at: datetime
    updated_at: datetime


class MenuPermissionMessageResponse(BaseModel):
    message: str
    data: MenuPermissionResponse | None = None
